compiler/rustc_parse_format/src/lib.rs

```diff
@@ -494,8 +494,11 @@ impl<'a> Parser<'a> {
             Some(ArgumentIs(i))
         } else {
             match self.cur.peek() {
-                Some(&(_, c)) if rustc_lexer::is_id_start(c) => {
-                    Some(ArgumentNamed(Symbol::intern(self.word())))
+                Some(&(start, c)) if rustc_lexer::is_id_start(c) => {
+                    let word = self.word();
+                    let end = self.cur.peek().map_or(0, |&(end, _)| end);
```

Comment:
```suggestion
                    let end = start + word.len();
```
?

It's simpler, and as written the `0` case looks incorrect (although it can probably only happen in ill-formed format strings).